packages/slate-react/src/components/content.js

```diff
@@ -161,20 +162,33 @@ class Content extends React.Component {
     }
 
     // If the new range matches the current selection, do nothing.
-    if (
-      current &&
-      range.startContainer == current.startContainer &&
-      range.startOffset == current.startOffset &&
-      range.endContainer == current.endContainer &&
-      range.endOffset == current.endOffset
-    ) {
-      return
+    if (current) {
+      if (
+        range.startContainer == current.startContainer &&
+        range.startOffset == current.startOffset &&
+        range.endContainer == current.endContainer &&
+        range.endOffset == current.endOffset
+      ) {
+        return
+      }
+      if (
+        range.startContainer == current.endContainer &&
+        range.startOffset == current.endOffset &&
+        range.endContainer == current.startContainer &&
+        range.endOffset == current.startOffset
+      ) {
+        return
+      }
```

Comment:
It seems weird to be doing both of these checks here.

Is the problem that the `range` returned from `findDOMRange` is properly set forward/backward depending on the Slate selection? If so wouldn't the fix be to fix the `findDOMRange` util instead?

Reply:
@ianstormtaylor I tried those first but then it actually get the start and end node correctly but ```range.setStart()``` and ```range.setEnd()``` resulting in the same range object whether it's forward/backward. that's why this check is needed